components/channel_header/channel_header.js

```diff
@@ -116,7 +116,7 @@ export default class ChannelHeader extends React.PureComponent {
         if (header !== prevHeader) {
             this.props.actions.getCustomEmojisInText(header);
         }
-        if (this.props.dmUser !== prevProps.dmUser && this.props.dmUser) {
+        if (this.props.dmUser !== prevProps.dmUser && this.props.dmUser && this.props.dmUser.is_bot) {
```

Comment:
`this.props.dmUser !== prevProps.dmUser` this looks like this can be called whenever `dmUser` object changes.Was the intention only to make the api call for different bot user?

Reply:
Right. That's why I made this change. So it's only called for bot users.

Reply:
My guess is that any property change on `dmUser`(Even when a bot) could trigger this. 
When we refresh the page on a DM now we see this executed multiple times hence the 404's. Would the sane happen if i refresh on a bot DM channel?

if we are only looking for a change looking for an ID change seems more robust `this.props.dmUser.id !== prevProps.dmUser.id`

Reply:
Yup that makes sense. Of course I have to do more checks for that because javascript :P